public/src/api/httpClient.ts

```diff
@@ -249,3 +249,15 @@ export function getFile(fileName: string): Promise<any> {
     }
   });
 }
+
+export function viewProduct(productName: string): Promise<any> {
+  return makeApiRequest({
+    url: `${ApiUrl.Products}/view_product`,
```

Comment:
Let's clarify the resource name

```suggestion
    url: `${ApiUrl.Products}/views-count`,
```